Feature/core 459 mnemonic export

karladler opened the pull request.

Commit:
add mnemonic-export and try to recover localStorage encrypted mnemonic

- [CORE-459]

karladler committed:
fix scss add basic mnemonic view

- [CORE-459] fix scss add basic mnemonic view
- fix some linter errors
- still WIP

datrinh committed:
Fixed some typos and wrong swapped translations

datrinh committed:
Added download text file functionality
Some refactoring

datrinh committed:
Added copy to clipboard function

datrinh committed:
Merge branch 'develop' into feature/CORE-459-mnemonic-export

# Conflicts:
#	dapps/evancore.vue.libs/src/i18n/de.json
#	dapps/evancore.vue.libs/src/i18n/en.json

datrinh committed:
Fixed missing comma

datrinh committed:
Added loggedin state to vuex
Added Modal as first draft

karladler committed:
add first rough print style for mnemonics

# Conflicts:
#	dapps/evancore.vue.libs/dbcp.json

datrinh committed:
Reverted style for notification bar

karladler committed:
fix print styles, add translations

karladler committed:
Add mnemonic export styles, print styles, pure-checkbox

- [CORE-459] add mnemonic export
- print styles for printing nice sheet
- dark styles according to screen design
- checkbox logic for acknowledge understanding importance of recovery key
- add pure checkbox without form-control

karladler committed:
comb CSS

karladler committed:
fix mnemonic oder count, fix FF print styles

karladler committed:
add go-secure / remove mnemonic function

Tschuck commented on dapps/evancore.vue.libs/src/components/dapp-wrapper/dapp-wrapper.ts:
```diff
@@ -119,6 +117,15 @@ export default class DAppWrapperComponent extends mixins(EvanComponent) {
   @Prop({ default: true }) createRuntime: boolean;
 
+  get isLoggedin() {
+    console.log('Logged In?', this.$store.state.isLoggedin);
```
please remove console logs. If you want `debug` logs, please use blockchain-core logging functions.
```suggestion
```

Tschuck commented on dapps/evancore.vue.libs/src/components/dapp-wrapper/dapp-wrapper.ts:
```diff
@@ -119,6 +117,15 @@ export default class DAppWrapperComponent extends mixins(EvanComponent) {
+  }
+  set isLoggedin(state) {
+    this.$store.commit('setLoginState', state);
+    console.log('Log State:', state);
```
please remove console logs. If you want `debug` logs, please use blockchain-core logging functions.
```suggestion
```

Tschuck reviewed (commented):
Their are a view differences between the implementation and the original wireframes. (design, alias is missing) Is it correct?

Notice: The checkbox for "I backuped my recovery key and understand the risk of loosing it." is very nice!

Current implementation:
![image](https://user-images.githubusercontent.com/7114819/68571619-b13c1680-0463-11ea-919a-835eaae4eacc.png)

Wireframes:
![image](https://user-images.githubusercontent.com/7114819/68571641-c0bb5f80-0463-11ea-90f2-4b6d31d9ddeb.png)

Tschuck commented:
please remove the old copy mnemonic dialog.

![image](https://user-images.githubusercontent.com/7114819/68571854-4808d300-0464-11ea-8328-c94927811f11.png)

Tschuck reviewed (changes requested):
please have a look at the comments

karladler committed:
Update dapps/evancore.vue.libs/src/components/dapp-wrapper/dapp-wrapper.ts

Co-Authored-By: Tschuck <tobias.winkler@evan.team>

karladler committed:
Update dapps/evancore.vue.libs/src/components/dapp-wrapper/dapp-wrapper.ts

Co-Authored-By: Tschuck <tobias.winkler@evan.team>

karladler commented:
> Their are a view differences between the implementation and the original wireframes. (design, alias is missing) Is it correct?
> 
> Notice: The checkbox for "I backuped my recovery key and understand the risk of loosing it." is very nice!
> 
> Current implementation:
> ![image](https://user-images.githubusercontent.com/7114819/68571619-b13c1680-0463-11ea-919a-835eaae4eacc.png)
> 
> Wireframes:
> ![image](https://user-images.githubusercontent.com/7114819/68571641-c0bb5f80-0463-11ea-90f2-4b6d31d9ddeb.png)

Alias is missing since it's taken from window local storage, but not written. This will be implemented within the re-login-refactoring branch.

karladler committed:
make header primary color

karladler committed:
Merge branch 'feature/CORE-459-mnemonic-export' of github.com:evannetwork/ui-vue into feature/CORE-459-mnemonic-export

karladler committed:
Merge branch 'develop' into feature/CORE-459-mnemonic-export

# Conflicts:
#	dapps/evancore.vue.libs/src/components/registry.ts

karladler committed:
improve txt download

- [CORE-459] use const instead of let
- remove obsolete backuplevel variable
- improve txt download file

karladler committed:
code styles

karladler committed:
fix indent

karladler commented on dapps/evancore.vue.libs/src/component.ts:
```diff
@@ -43,6 +38,8 @@ export default class EvanComponent extends Vue {
    */
   dapp: any;
 
+  dispatcher: any;
```
```suggestion
  dispatcher;
```

karladler committed:
Update dapps/evancore.vue.libs/src/component.ts

karladler commented on dapps/evancore.vue.libs/src/i18n/en.json:
```diff
@@ -67,6 +67,20 @@
+      "print": "Print",
+      "recovery-key": "Recovery Key",
+      "title": "Print your Recovery Key",
+      "understood": "I backuped my recovery key and understand the risk of loosing it."
```
```suggestion
      "understood": "I backed up my recovery key and understand the risk of loosing it."
```

karladler committed:
Update dapps/evancore.vue.libs/src/i18n/en.json

karladler commented on dapps/evancore.vue.libs/src/i18n/en.json:
```diff
@@ -67,6 +67,20 @@
+      "account-id": "Account ID",
+      "alias": "Alias",
+      "created-at": "Created at:",
+      "description": "Your account is protected by your password and recovery key. You will need your recovery key to log in from new devices.\n\nPlease note: The recovery key is not stored in your profile information and could not be set back. In case of lost, you will loose access to your business partners, contacts and profile information.",
```
```suggestion
      "description": "Your account is protected by your password and recovery key. You will need your recovery key to log in from new devices.\n\nPlease note: The recovery key is not stored in your profile information and could not be set back. In case of lost, you will lose access to your business partners, contacts and profile information.",
```

karladler committed:
Update dapps/evancore.vue.libs/src/i18n/en.json

karladler commented on dapps/evancore.vue.libs/src/components/dapp-wrapper/mnemonic-export/mnemonic-export.vue:
```diff
@@ -0,0 +1,118 @@
+        <div class="print-view">
+          <div class="row">
+            <!-- dark left info box -->
+            <div class="col-md-6 info-box inverted">
```
```suggestion
            <div class="col-md-6 info-box inverted pr-7">
```

karladler committed:
Update dapps/evancore.vue.libs/src/components/dapp-wrapper/mnemonic-export/mnemonic-export.vue

add paddinf rigth to infobox

karladler committed:
add toast when copied mnemonic

Commit:
fix i18n

- [CORE-459]

Commit:
Merge branch 'develop' into feature/CORE-459-mnemonic-export
